main.tf

```diff
@@ -76,11 +79,6 @@ resource "azurerm_network_interface" "nic" {
   }
 }
 
-resource "azurerm_network_interface_security_group_association" "nic-nsg" {
```

Comment:
Why did you remove this?

Reply:
I believe there was a dependency change in the az provider because it is not a requirement.